jax/_src/numpy/polynomial.py

```diff
@@ -102,15 +115,131 @@ def roots(p: ArrayLike, *, strip_zeros: bool = True) -> Array:
-Unlike NumPy's implementation of polyfit, :py:func:`jax.numpy.polyfit` will not warn on rank reduction, which indicates an ill conditioned matrix
-Also, it works best on rcond <= 10e-3 values.
-"""
-@implements(np.polyfit, lax_description=_POLYFIT_DOC)
 @partial(jit, static_argnames=('deg', 'rcond', 'full', 'cov'))
 def polyfit(x: Array, y: Array, deg: int, rcond: float | None = None,
             full: bool = False, w: Array | None = None, cov: bool = False
             ) -> Array | tuple[Array, ...]:
+  r"""Least squares polynomial fit to data.
+
+  Jax implementation of :func:`numpy.polyfit`.
+
+  Given a set of data points ``(x, y)`` and degree of polynomial ``deg``, the
+  function finds a polynomial equation of the form:
+
+  .. math::
+
+	   y = p(x) = p[0] x^{deg} + p[1] x^{deg - 1} + ... + p[deg]
+
+  Args:
+    x: Array of data points of shape ``(M,)``.
+    y: Array of data points of shape ``(M,)`` or ``(M, K)``.
+    deg: Degree of the polynomials.
+    rcond: Relative condition number of the fit. Default value is ``len(x) * eps``.
+    full: Switch that controls the return value. Default is ``False`` which
+      restricts the return value to the array of polynomail coefficients ``p``.
+      If ``True``, the function returns a tuple ``(p, resids, rank, s, rcond)``.
+    w: Array of weights of shape ``(M,)``. If None, all data points are considered
+      to have equal weight. If not None, the weight :math:`w_i` is applied to the
+      unsquared residual of :math:`y_i - \widehat{y}_i` at :math:`x_i`, where
+      :math:`\widehat{y}_i` is the fitted value of :math:`y_i`. Default is None.
+    cov: Boolean or string. If ``True``, returns the covariance matrix scaled
+      by ``resids/(M-deg-1)`` along with ploynomial coefficients. If
+      ``cov='unscaled'``, returns the unscaaled version of covariance matrix.
+      Default is ``False``. ``cov`` is ignored if ``full=True``.
+
+  Returns:
+    - An array polynomial coefficients ``p`` if ``full=False`` and ``cov=False``.
+
+    - A tuple of arrays ``(p, resids, rank, s, rcond)`` if ``full=True``. Where
+
+      - ``p`` is an array of shape ``(M,)`` or ``(M, K)`` containing the polynomial
+        coefficients.
+      - ``resids`` is the sum of squared residual of shape () or (K,).
+      - ``rank`` is the rank of the matrix ``x``.
+      - ``s`` is the singular values of the matrix ``x``.
+      - ``rcond`` as the array.
+    - A tuple of arrays ``(p, C)`` if ``full=False`` and ``cov=True``. Where
+
+      - ``p`` is an array of shape ``(M,)`` or ``(M, K)`` containing the polynomial
+        coefficients.
+      - ``C`` is the covariance matrix of polynomial coefficients of shape
+        ``(deg + 1, deg + 1)`` or ``(deg + 1, deg + 1, 1)``.
+
+  Note:
+    Unlike :func:`numpy.polyfit` implementation of polyfit, :func:`jax.numpy.polyfit`
+    will not warn on rank reduction, which indicates an ill conditioned matrix.
+    Also, it works best on rcond <= 10e-3 values.
```

Comment:
Sorry, just noticed this. What does "works best on rcond <= 10e-3 values" mean? Does it mean that the user should not use the default rcond?

Reply:
Default rcond is <= 10e-3. It does not mean that the user should not use the default rcond. If the user wants to change the dafault rcond value, then it suggested to use the value <= 10e-3.

Reply:
I see - it looks like this comes from the original PR (#7751). I don't entirely understand where the `10e-3` number comes from (or whether it's meant to be `0.01` or `0.001`). I think this adds more confusion than clarity, and we should remove mention of it.